[Custom Fields] Empty state

hichamboushaba opened the pull request:
<!-- Remember about a good descriptive title. -->

Closes: #12732
<!-- Id number of the GitHub issue this PR addresses. -->

### Description
This PR adds an empty state to the custom fields screen.

It also updates the logic of the top banner to be shown only when there are pending changes, this was discussed here p1727697084758769/1727685068.374159-slack-C03L1NF1EA3

### Steps to reproduce
##### Orders
1. Open an order in the app.
2. Tap on "Edit Fields" button.
3. If there are some existing fields, then delete them.

##### Products
1. Open a product in the app.
2. Tap on "Add more details", then select custom fields.

### Testing information
- Check the UI of the empty view.
- Test the Learn more button.
- Confirm the empty view is hidden when adding some custom fields.
- Confirm the top banner is shown when there are some changes to save.

### The tests that have been performed
The above.

### Images/gif
<img src=https://github.com/user-attachments/assets/0aeb4fc1-5e2f-4e5a-87d7-d07beae2f72c width=360 />

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [x] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [x] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [x] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on big (tablet) and small (phone) in case of UI changes, and no regressions are added.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->

hichamboushaba committed:
Add the empty state UI

hichamboushaba committed:
Handle the learn more button

hichamboushaba committed:
Add a unit test

hichamboushaba committed:
Show the top banner only when the save button is shown

dangermattic commented:
<!--
  0 Errors
  1 Warning: This PR is assigned to the mil...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-android/milestone/239">20.7</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

JorgeMucientes commented on WooCommerce/src/test/kotlin/com/woocommerce/android/ui/customfields/list/CustomFieldsViewModelTest.kt:
```diff
@@ -408,20 +411,23 @@ class CustomFieldsViewModelTest : BaseUnitTest() {
-        testBlocking {
-            appPrefs.isCustomFieldsTopBannerDismissed = false
-            setup()
+    fun `given custom fields top banner is not dismissed, when screen is opened, then banner is shown`() = testBlocking {
```
Minor np. Wouldn't this now be `when there are changes to custom fields` rather than `when the screen is opened` ?

hichamboushaba replied:
Good point, I'll update it.

JorgeMucientes reviewed (commented):
Great work @hichamboushaba the code looks good and for the most part works well, but I found an issue that should be addressed before merging: 

When entering custom fields screen: 
1. Remove existing custom fields
2. See the empty case
3. Save
4. Now try to add new custom fields
5. The empty case is still shown; the user can't save the changes or navigate back; it is stuck. 

https://github.com/user-attachments/assets/9bb79b78-fe9a-4538-91da-1be8b10dbb22

hichamboushaba commented:
> Great work @hichamboushaba the code looks good and for the most part works well, but I found an issue that should be addressed before merging:

Thanks @JorgeMucientes for the review, I tried replicating the issue but I can't.
Can you please confirm if the branch on your laptop is up-to-date, the PR was based on top of `issue/12699-json-fields-read-only` initially, and when I created it I found some issue with the JSON overlay logic that I fixed later, then when the previous PR was merged, I rebased the current branch, and that's the only think I can think of that could cause the issue.

hichamboushaba committed:
Correct name of the unit test

JorgeMucientes reviewed (approved):
Hey @hichamboushaba I'm not sure what was happening before. I did a git pull on the local branch and everything seemed `Already up to date.` according to Git. In any case, after a clean install I couldn't reproduce the issue anymore 🤔.

Approving since everything is working fine in my recent tests 👍🏼
